DllImportGenerator/DllImportGenerator/Analyzers/ConvertToGeneratedDllImportAnalyzer.cs

```diff
@@ -0,0 +1,125 @@
+                        return;
+
+                    compilationContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext, dllImportAttrType), SymbolKind.Method);
+                });
+        }
+
+        private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol dllImportAttrType)
+        {
+            var method = (IMethodSymbol)context.Symbol;
+
+            // Check if method is a DllImport
+            DllImportData? dllImportData = method.GetDllImportData();
+            if (dllImportData == null)
+                return;
+
+            // Ignore QCalls
+            if (dllImportData.ModuleName == "QCall")
+                return;
+
+            if (RequiresMarshalling(method, dllImportData, dllImportAttrType))
+            {
+                context.ReportDiagnostic(method.CreateDiagnostic(ConvertToGeneratedDllImport, method.Name));
+            }
+        }
+
+        private static bool RequiresMarshalling(IMethodSymbol method, DllImportData dllImportData, INamedTypeSymbol dllImportAttrType)
+        {
+            // SetLastError=true requires marshalling
+            if (dllImportData.SetLastError)
+                return true;
+
+            // Check if return value requires marshalling
+            if (!method.ReturnsVoid && RequiresMarshalling(method.ReturnType))
+                return true;
+
+            // Check if parameters require marshalling
+            foreach (IParameterSymbol paramType in method.Parameters)
+            {
+                if (paramType.RefKind != RefKind.None)
+                    return true;
+
+                if (RequiresMarshalling(paramType.Type))
+                    return true;
+            }
+
+            // DllImportData does not expose all information (e.g. PreserveSig), so we still need to get the attribute data
+            AttributeData? dllImportAttr = null;
+            foreach (AttributeData attr in method.GetAttributes())
+            {
+                if (!SymbolEqualityComparer.Default.Equals(attr.AttributeClass, dllImportAttrType))
+                    continue;
+
+                dllImportAttr = attr;
+                break;
+            }
+
+            Debug.Assert(dllImportAttr != null);
+            foreach (var namedArg in dllImportAttr!.NamedArguments)
+            {
+                if (namedArg.Key != nameof(DllImportAttribute.PreserveSig))
+                    continue;
+
+                // PreserveSig=false requires marshalling
+                if (!(bool)namedArg.Value.Value!)
+                    return true;
+            }
+
+            return false;
+        }
+
+        private static bool RequiresMarshalling(ITypeSymbol typeSymbol)
+        {
+            if (typeSymbol.TypeKind == TypeKind.Enum)
+                return false;
+
+            if (typeSymbol.TypeKind == TypeKind.Pointer)
+                return false;
+
+            return !TypeSymbolExtensions.IsSpecialTypeBlittable(typeSymbol.SpecialType);
```

Comment:
You should be able to just call `IsTypeConsideredBlittable` instead of handling the SpecialType separately.